sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -2,9 +2,12 @@
 // Licensed under the MIT License.
 
 using System.ClientModel.Internal;
+using System.ClientModel.Options;
+using System.ClientModel.Primitives;
 using System.Collections;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
```

Comment:
Why Linq?  These calls typically have a perf cost that's better to avoid, so if there's another way to achieve the same goal, that might be preferred.

Reply:
Yeah I was using .ToArray() on the allowed headers/queryparameters, I'll try out removing this as part of freezing them in the options class using ReadOnly